### DISCUSSION TITLE
Save parameter issue ?

### DISCUSSION THREAD

**Discussion by rolfdegen** [ElectroTechnique/electrotechnique.github.io#27] [Software Issues]
Detune issue in Osc2
Smaller detune values are not saved correctly. Can someone check that ?

This is the Detune update in TSynth:
![Detune](https://user-images.githubusercontent.com/16689445/112315634-9958e180-8caa-11eb-8088-5353820cd00c.JPG)

**Comment by rolfdegen** [comment]
Only 2 decimal places are saved. This is not enough for a small oscillator tune !?

**Reply by rolfdegen** [reply]
I am currently testing glideSpeed parameters. It's the same problem with detune. The Patch parameter on SDCard has only two decimal places. That is not enough for small values under 14.

The calculation is gliedeSpeed = POWER[Value]. 

The POWER table has very small numbers with 5 decimal places. Example glideSpeed Midi value is 7. The result is float 0.003. Saved is 0.00. That confuses me !?

![POWERTab](https://user-images.githubusercontent.com/16689445/112484786-89a8ce00-8d7a-11eb-8c15-7c08979b556b.JPG)

**Reply by rolfdegen** [reply]
I adjusted my code for 5 decimal digits and it works fine.

![Code](https://user-images.githubusercontent.com/16689445/112493833-ce386780-8d82-11eb-8e0f-42c2cbb1c498.JPG)

**Comment by Electrotechnique** [comment]
This will be fixed in the next release. 

`  return patchName + "," + String(oscALevel) + "," + String(oscBLevel) + "," + String(noiseLevel) + "," + String(unison) + "," + String(oscFX) + "," + String(detune,5) + "," + String(lfoSyncFreq) + "," + String(midiClkTimeInterval) + "," + String(lfoTempoValue) + "," + String(keytrackingAmount) + "," + String(glideSpeed,5) + "," + String(oscPitchA) + "," + String(oscPitchB) + "," + String(oscWaveformA) + "," + String(oscWaveformB) + "," +
         String(pwmSource) + "," + String(pwmAmtA) + "," + String(pwmAmtB) + "," + String(pwmRate) + "," + String(pwA) + "," + String(pwB) + "," + String(filterRes) + "," + String(filterFreq) + "," + String(filterMix) + "," + String(filterEnv) + "," + String(oscLfoAmt,5) + "," + String(oscLfoRate,5) + "," + String(oscLFOWaveform) + "," + String(oscLfoRetrig) + "," + String(oscLFOMidiClkSync) + "," + String(filterLfoRate,5) + "," +
         filterLfoRetrig + "," + filterLFOMidiClkSync + "," + filterLfoAmt + "," + filterLfoWaveform + "," + filterAttack + "," + filterDecay + "," + filterSustain + "," + filterRelease + "," + ampAttack + "," + ampDecay + "," + ampSustain + "," + ampRelease + "," +
         String(fxAmt) + "," + String(fxMix) + "," + String(pitchEnv) + "," + String(velocitySens) + "," + String(chordDetune) + "," + String(0.0f) + "," + String(0.0f) + "," + String(0.0f);`

**Comment by rolfdegen** [comment]
Fine. Thanks :)

**Comment by Electrotechnique** [comment]
Fixed in V2.11
